tensorflow_datasets/testing/mocking.py

```diff
@@ -104,18 +106,34 @@ def mock_download_and_prepare(self, *args, **kwargs):
           ''.format(self._data_dir, n=self.name)  # pylint: disable=protected-access
       )
 
-  def mock_as_dataset(self, *args, **kwargs):
+  def mock_as_dataset(self, split, decoders=None, **kwargs):
     """Function which overwrite builder._as_dataset."""
-    del args
+    del split
     del kwargs
+
+    features = self.info.features
+    # `from_generator` takes a callable with signature () -> iterable
+    # Recreating a new generator each time ensure that all pipelines are
+    # using the same examples
+    def generator():
+      for ex in RandomFakeGenerator(builder=self, num_examples=num_examples):
+        if decoders:
+          ex = features.encode_example(ex)
```

Comment:
Unfortunately we cannot always encoded examples which would simplify the code, since it would fail for datasets using `Text` features, due to the way how the fake data is generated.

Reply:
Could you provide more context ? Why text encoding fails when encoding the random data ?

Reply:
Sorry for the unclear description. Random data for the text features are generated as a list of random integers depending on the vocab size. However for encoding, the text feature expects strings as it often has a custom additional encoder to convert the text into integers encoding the vocabulary:
https://github.com/tensorflow/datasets/blob/2fc35db14c418e3c5415bb3511fdecabb8a1b252/tensorflow_datasets/core/features/text_feature.py#L111-L114

Reply:
Thank for the explanations, that make sense. The encoder API is deprecated so hopefully it shouldn't really matter in this case.
I'm thinking but don't think there are other instance where `encode_example` expect a different format than what is returned by `decode_example`.
I'll try to simplify the code when merging this internally.

Reply:
Thanks for looking into it.